content/blog/asyncapi_codegen_scst.md

```diff
@@ -0,0 +1,136 @@
+
+[![asyncapi playground for creating microservices using spring cloud stream](https://solace.com/wp-content/uploads/2020/04/asyncapi-codegen_pic-01-1024x499.png)](https://solace.com/wp-content/uploads/2020/04/asyncapi-codegen_pic-01.png)
+
+The second way is to use an event portal. Solace PubSub+ Event Portal, for example, allows for architects and developers to collaborate using a GUI to design your event-driven architecture. The team would define the applications that exist in the system, as well as the events that are exchanged and the schemas which define them. Having a catalog of well-organized channels and events for reuse will also save you both time and headaches while collaborating, instead of having to comb through a bunch of files in various locations.
+
+Once the design is in place, PubSub+ Event Portal allows the developer to choose the application they are responsible for developing and download the AsyncAPI document in JSON or YAML.
+
+**[![](https://solace.com/wp-content/uploads/2020/04/asyncapi-codegen_pic-02-1024x503.png)](https://solace.com/wp-content/uploads/2020/04/asyncapi-codegen_pic-02.png) **
+
+# **Create Event-Driven Microservices Using Spring Cloud Stream Without Learning Messaging APIs**
+
+Now that we have our AsyncAPI document that describes our application it’s time to develop the application. The AsyncAPI [Code Generator](https://github.com/asyncapi/generator) supports templates to generate code for a variety of different languages and protocols, but for this example we’re going to use the [Spring Cloud Stream template](https://github.com/asyncapi/java-spring-cloud-stream-template). One should note that the template generates a Maven project.
+
+The Spring Cloud Stream framework provides an easy way to get started with event-driven microservices by providing binders that allow the developer to create their microservices without having to learn messaging APIs.
+
+### **Download and Run the AsyncAPI Generator**
+
+The first step is of course to install the AsyncAPI generator itself. If you have NodeJS installed this takes just one easy `npm` command as seen below. You can find the required versions in the [Code Generator](https://github.com/asyncapi/generator) on github.
+
+```
+npm install -g @asyncapi/generator
+```
+
+Once you have the generator installed you can run it using the `ag` command. At a minimum you must specify the AsyncAPI document to run it against and the template to use as shown below.
+
+```
+ag ~/AsyncApiDocument.yaml https://github.com/asyncapi/java-spring-cloud-stream-template.git
```

Comment:
better do `ag ~/AsyncApiDocument.yaml @asyncapi/java-spring-cloud-stream-template` as this template is not always released to https://www.npmjs.com/package/@asyncapi/java-spring-cloud-stream-template and releases are automated, so whenever there is a feature or a fix, it goes immediately to npm. If you do it this way, the advantage is that you can also secure to use some released version only, to not get unexpected errors that might come from newer versions (so surprises during demos :))

What I mean is that you can do `ag ~/AsyncApiDocument.yaml @asyncapi/java-spring-cloud-stream-template@0.3.2` for example